services/system/Transact/src/Transact.cpp

```diff
@@ -93,6 +93,41 @@ namespace SystemService
             break;
          includedTable.remove(obj);
       }
+
+      auto snap    = tables.open<SnapshotInfoTable>();
+      auto snapRow = snap.get(SingletonKey{});
+      if (snapRow)
+      {
+         if (snapRow->snapshotInterval &&
+             stat.current.time.seconds - snapRow->lastSnapshot.seconds >= snapRow->snapshotInterval)
+         {
+            ScheduledSnapshotRow row{stat.current.blockNum};
+            kvPut(ScheduledSnapshotRow::db, row.key(), row);
+            snapRow->lastSnapshot = stat.current.time;
+            snap.put(*snapRow);
+         }
+      }
+      else
+      {
+         snap.put({stat.head->header.time, 0});
+      }
+   }
+
+   void Transact::setSnapTime(std::uint32_t seconds)
```

Comment:
- 0 is special-cased to mean no snapshot
- 1 will probably be very slow, but shouldn't cause any other problems.

How small it can be without killing performance depends on the size of the state.

Reply:
> 0 is special-cased to mean no snapshot

Should be documented